File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], K: int) -> Optional[ListNode]:
        if K==0:
            return head
        temp=head
        cnt=0
        while temp:
            temp=temp.next
            cnt+=1
        if cnt==0:
            return
        start=head
        K=K%cnt
        k=(cnt-K)-1
        while k:
            head=head.next
            k-=1
        temp1=head
        while head.next:
            head=head.next
        head.next=start
        ans=temp1.next
        temp1.next=None
        return ans

File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(head==NULL || head->next==NULL || k==0){
            return head;
        }
        int cnt=1;
        ListNode* temp=head;
        while(temp->next!=NULL){
            cnt=cnt+1;
            temp=temp->next;
        }
        temp->next=head;
        cout<<cnt<<endl;
        k=k%cnt;
        int K=cnt-k-1;
        cout<<K<<endl;
        ListNode* temp2=head;
        while (K>0){
            temp2=temp2->next;
            K-=1;
        }
        ListNode* ans=temp2->next;
        temp2->next=NULL;
        return ans;
    }
};